pallets/stable-staking/src/lib.rs

```diff
@@ -0,0 +1,888 @@
+			);
+			ensure!(
+				!StakingPoolSetting::<T>::contains_key(&pool_id),
+				Error::<T>::PoolAlreadyExisted
+			);
+			<StakingPoolSetting<T>>::insert(pool_id.clone(), setting.clone());
+			Self::deposit_event(Event::StakingPoolCreated {
+				pool_id,
+				start_time: setting.start_time,
+				epoch: setting.epoch,
+				epoch_range: setting.epoch_range,
+				setup_time: setting.setup_time,
+				pool_cap: setting.pool_cap,
+			});
+			Ok(())
+		}
+
+		// name = None will cause remove of metadata
+		#[pallet::call_index(1)]
+		#[pallet::weight({1000})]
+		pub fn update_metadata(
+			origin: OriginFor<T>,
+			pool_id: T::PoolId,
+			name: Option<Vec<u8>>,
+			description: Vec<u8>,
+		) -> DispatchResult {
+			T::StakingPoolCommitteeOrigin::ensure_origin(origin)?;
+			ensure!(StakingPoolSetting::<T>::contains_key(&pool_id), Error::<T>::PoolNotExisted);
+			if let Some(name_inner) = name {
+				let bounded_name: BoundedVec<u8, T::PoolStringLimit> =
+					name_inner.clone().try_into().map_err(|_| Error::<T>::BadMetadata)?;
+
+				let bounded_description: BoundedVec<u8, T::PoolStringLimit> =
+					description.clone().try_into().map_err(|_| Error::<T>::BadMetadata)?;
+				<StakingPoolMetadata<T>>::insert(
+					pool_id.clone(),
+					PoolMetadata { name: bounded_name, description: bounded_description },
+				);
+				Self::deposit_event(Event::MetadataSet { pool_id, name: name_inner, description });
+			} else {
+				<StakingPoolMetadata<T>>::remove(pool_id.clone());
+				Self::deposit_event(Event::MetadataRemoved { pool_id });
+			}
+			Ok(())
+		}
+
+		/// Removes a resource ID from the resource mapping.
+		///
+		/// After this call, bridge transfers with the associated resource ID will
+		/// be rejected.
+		#[pallet::call_index(2)]
+		#[pallet::weight({1000})]
+		#[transactional]
+		pub fn update_reward(
+			origin: OriginFor<T>,
+			pool_id: T::PoolId,
+			epoch: u128,
+			reward: BalanceOf<T>,
+		) -> DispatchResult {
+			T::StakingPoolCommitteeOrigin::ensure_origin(origin)?;
+
+			let asset_id = <AIUSDAssetId<T>>::get().ok_or(Error::<T>::NoAssetId)?;
+			let actual_reward = T::Fungibles::mint_into(
+				asset_id,
+				&Self::stable_token_beneficiary_account(),
+				reward,
+			)?;
+
+			<StableStakingPoolEpochReward<T>>::try_mutate(
+				&pool_id,
+				&epoch,
+				|maybe_reward| -> DispatchResult {
+					ensure!(maybe_reward.is_none(), Error::<T>::RewardAlreadyExisted);
+
+					*maybe_reward = Some(StableRewardInfo { epoch, reward_amount: actual_reward });
+					Self::deposit_event(Event::<T>::RewardUpdated {
+						pool_id: pool_id.clone(),
+						epoch,
+						amount: actual_reward,
+					});
+					let current_block = frame_system::Pallet::<T>::block_number();
+					let current_epoch = Self::get_epoch_index(pool_id.clone(), current_block)?;
+					ensure!(current_epoch <= epoch, Error::<T>::EpochAlreadyEnded);
+					Ok(())
+				},
+			)?;
+
+			<StableStakingPoolReward<T>>::try_mutate(&pool_id, |maybe_reward| -> DispatchResult {
```

Comment:
How about name it `reward` or `pool_reward`? as `StableStakingPoolReward` uses `ValueQuery`

Reply:
Done